internal/host/statemachine.go

```diff
@@ -255,16 +257,69 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th *transitionHandler) sta
 		PostTransition:   th.PostUnbindHost,
 	})
 
-	// Reclaim host
+	// ReclaimHost when installed moves to Reclaiming
 	sm.AddTransition(stateswitch.TransitionRule{
 		TransitionType: TransitionTypeReclaimHost,
 		SourceStates: []stateswitch.State{
 			stateswitch.State(models.HostStatusInstalled),
+			stateswitch.State(models.HostStatusAddedToExistingCluster),
 		},
 		DestinationState: stateswitch.State(models.HostStatusReclaiming),
 		PostTransition:   th.PostUnbindHost,
 	})
 
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRebootingForReclaim,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusReclaiming),
+		},
+		DestinationState: stateswitch.State(models.HostStatusReclaimingRebooting),
+	})
+
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeReclaimFailed,
```

Comment:
who is going to use this transition?

Reply:
This would be used when the agent returns an error from one of the steps in this process.

Reply:
handling get step reply?

Reply:
Yes that's the case, but that handling isn't being implemented in this PR. I just opened https://issues.redhat.com/browse/MGMT-11335 yesterday for that work.

Reply:
I intended this PR to just be the states and transitions.

I added the monitor stuff because it was fairly minor (two lines and some tests), but I can separate that out if you'd prefer?

Reply:
I also updated the diagram in the enhancement to reflect this state's usage and the monitor change, hopefully that helps.